src/js/modules/ControllerManager/Keypad.js

```diff
@@ -0,0 +1,220 @@
+ *
+ * The Foundry VTT Module Mindflayer is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * The Foundry VTT Module Mindflayer is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with the Foundry VTT Module Mindflayer. If not,
+ * see <https://www.gnu.org/licenses/>.
+ */
+"use strict";
+import { LOG_PREFIX, VTT_MODULE_NAME } from "../../settings/constants";
+import { Key } from "./Key";
+import * as TokenUtil from "../../utils/tokenUtil";
+
+/**
+ * Keypad class representing a keypad
+ */
+export default class Keypad {
+  #instance;
+  #controllerId;
+  #rawState = {
+    Q: new Key(),
+    W: new Key(),
+    E: new Key(),
+    A: new Key(),
+    S: new Key(),
+    D: new Key(),
+    Z: new Key(),
+    X: new Key(),
+    C: new Key(),
+    SHI: new Key(),
+    SPC: new Key(),
+  };
+  #ledChanged = false;
+  #ledState = ["#000000", "#000000"];
+
+  constructor(instance, controllerId) {
+    this.#instance = instance;
+    this.#controllerId = controllerId;
+    this.setDefaultLEDColor();
+    console.debug(
+      LOG_PREFIX + "Initialized keypad for controller " + this.#controllerId
+    );
+  }
+
+  get controllerId() {
+    return this.#controllerId;
+  }
+
+  get rotation() {
+    let r = game.user.getFlag(
+      VTT_MODULE_NAME,
+      "controllerRotation_" + this.#controllerId
+    );
+    if (r === undefined) {
+      r = 0;
+      this.rotation = r;
+    }
+    return r;
+  }
+
+  set rotation(amount) {
+    if (typeof amount !== "number") {
+      throw new TypeError("Rotation has to be numerical");
+    }
+    amount = amount % 360;
+    if (amount < 0) {
+      amount += 360;
+    }
+    game.user.setFlag(
+      VTT_MODULE_NAME,
+      "controllerRotation_" + this.#controllerId,
+      amount
+    );
+  }
+
+  /**
+   * Returns an assigned player for this keypad.
+   *
+   * @returns {Player | null} the assigned player
+   */
+  get player() {
+    const settings = this.#instance.settings.settings;
+    const playerId = Object.keys(settings.mappings).find(
+      (key) => settings.mappings[key] == this.#controllerId
+    );
+    const selectedPlayer = game.users.contents.find(
+      (player) => player.id == playerId
+    );
+    if (!selectedPlayer) {
+      return null;
+    }
+    return selectedPlayer;
+  }
+
+  /**
+   * Returns the currently selected Token for this keypad
+   *
+   * @returns {Token | null} the assinged token
+   */
+  get token() {
+    const player = this.player;
+    if (!player) return null;
+    return TokenUtil.getTokenFor(player, true);
+  }
+
+  getKeys() {
```

Comment:
Should we use the same syntax for every get function?
```suggestion
  get keys() {
```

Reply:
good idea, though that will need to be changed in the other files as well.